cypress/e2e/Library/datasets_remote.cy.js

```diff
@@ -0,0 +1,141 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Xibo.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* eslint-disable max-len */
+describe('Datasets', function() {
+  let testRun = '';
+
+  beforeEach(function() {
+    cy.login();
+
+    testRun = Cypress._.random(0, 1e9);
+  });
+
+  // ---------
+  // Tests - Error handling
+  it('should not add a remote dataset without URI', function() {
+    cy.visit('/dataset/view');
+
+    // Click on the Add Dataset button
+    cy.contains('Add DataSet').click();
+
+    cy.get('.modal input#dataSet')
+      .type('Cypress Test Dataset ' + testRun);
+
+    cy.get('.modal input#isRemote').check();
+
+    // Add first by clicking next
+    cy.get('.modal .save-button').click();
+
+    // Click on the "Remote" tab
+    cy.get(':nth-child(2) > .nav-link').should('be.visible').click();
+
+    // Check that the error message is displayed for the missing URI field
+    cy.get('#uri-error').should('have.text', 'This field is required.');
+  });
+
+  it('should be able to add json remote dataset', function() {
+    cy.visit('/dataset/view');
+
+    // Click on the Add Dataset button
+    cy.contains('Add DataSet').click();
+
+    cy.get('.modal input#dataSet')
+      .type('Cypress Test Dataset ' + testRun + '_1');
+
+    cy.get('#isRemote').click();
+
+    cy.get('a[href="#gateway"]').click();
+
+    cy.get('#uri').type('https://ceu.xibo.co.uk/test.json');
+
+    cy.get('a[href="#data"]').click();
+
+    cy.get('#dataRoot').type('data');
+
+    cy.get('#dataSetRemoteTestButton').click();
+
+    // Add first by clicking next
+    cy.get('.modal .save-button').click();
+
+    // Check if dataset is added in toast message
+    cy.contains('Added Cypress Test Dataset ' + testRun + '_1');
+  });
+  
+  it('should be able to add csv remote dataset', function() {
+    cy.visit('/dataset/view');
+
+    // Click on the Add Dataset button
+    cy.contains('Add DataSet').click();
+
+    cy.get('.modal input#dataSet')
+      .type('Cypress Test Dataset ' + testRun + '_1');
+
+    cy.get('#isRemote').click();
+
+    cy.get('a[href="#gateway"]').click();
+
+    cy.get('#uri').type('http://localhost/cy_people.csv');
+
+    cy.get('a[href="#data"]').click();
+
+    cy.get('#sourceId').select(1);
+
+    // cy.get('#dataRoot').type('data');
+
+    cy.get('#dataSetRemoteTestButton').click();
+
+    // Add first by clicking next
+    cy.get('.modal .save-button').click();
+
+    // Check if dataset is added in toast message
+    cy.contains('Added Cypress Test Dataset ' + testRun + '_1');
+  });
+
+  it.skip('should be seeing data in created remote dataset', function() {
+    cy.visit('/dataset/view');
+
+    cy.get('div[title="Row Menu"]').first().click();
+    
+    cy.contains('View Data').click();
+
+    cy.get('tbody tr').should('have.length.greaterThan', 1)
+  });
+  
+  it('should not be able to add remote dataset if no URI is populated', function() {
```

Comment:
What’s the difference between this test and the first test? They look similar to me.